File: Coding Exercises/Python/Day20/are_anagrams.py
def are_anagrams(str1, str2):
    str1 = str1.replace(" ", "").lower()
    str2 = str2.replace(" ", "").lower()

    if len(str1) != len(str2):
        return False

    char_list1 = list(str1)
    char_list2 = list(str2)

    char_list1.sort()
    char_list2.sort()

    return char_list1 == char_list2

File: Coding Exercises/TypeScript/Day20/areAnagrams.ts
function areAnagrams(str1: string, str2: string): boolean {
  str1 = str1.replace(/\s/g, '').toLowerCase();
  str2 = str2.replace(/\s/g, '').toLowerCase();

  if (str1.length !== str2.length) {
      return false;
  }

  const charArray1 = str1.split('');
  const charArray2 = str2.split('');

  charArray1.sort();
  charArray2.sort();

  return charArray1.join('') === charArray2.join('');
}
